### PR TITLE
Add exponential backoff to page_server->send

### PR DESCRIPTION
**Pull request by save-buffer**
## Problem
We currently spin and send a lot of stuff to hte pageserver if something goes wrong. Tracked in #6043 and #5757

## Summary of changes
Adds an exponential backoff to these connection attempts.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by knizhnik — COMMENTED** [review]
I do not think that this two place is right place to add exponential backoff.
There are more places where reconnection can be performed in loop (for example `prefetch_wait_for`)

It seem to be more natural to move this logic to connect.

**Review by koivunej — APPROVED** [review]
This looks good to me assuming these loops have a way out with the connection being closed and are then retried at another level, for example if pageserver connectionstring is changed?

File: pgxn/neon/pagestore_smgr.c
Comment (jcsp) [inline pgxn/neon/pagestore_smgr.c:265]: It's useful to name variables with a unit suffix when it's not obvious (I guess `_US` in these cases?)

**Comment by koivunej** [comment]
I think this has now evolved to what @knizhnik was after, leaving the approval to them.

**Comment by jcsp** [comment]
It seems like we still don't have a common understanding that exponential backoff is a good thing, but we need to get rid of the spinnning behavior ASAP.  @save-buffer can you update this PR's description to reflect what it does (current title is out of date).  We can keep the ticket open for implementing exponential backoff, but having some kind of delay between connection attempts is still better than nothing, so let's not hold this up.

**Comment by save-buffer** [comment]
I think I've found a middle-ground. I don't think @knizhnik's issue was with exponential backoff per se, but with where in the code it was implemented. Implementing it inside of `pageserver_connect` does have advantages, and so he proposed a rate-limiting approach. I understand we'd want to do quick retries early on, so I've switched to an "exponential rate-limiting" approach, where if we retry our connection before the previous delay has elapsed, we double our delay. I _think_ this is a pretty elegant solution with the properties we want.

**Review by knizhnik — CHANGES_REQUESTED** [review]
I wonder why do we need reconnect interval smaller than 1 second?
Are there many cases when there is some short-time network outage (network blinking) and connection between compute and PS should be reestablished ASAP?

File: pgxn/neon/libpagestore.c
Comment (save-buffer) [reply pgxn/neon/libpagestore.c]: I thoguht I'd tried to compile it... Fixed

**Review by knizhnik — APPROVED** [review]

**Pull request merged**
